app/src/main/java/org/thoughtcrime/securesms/home/search/GlobalSearchViewModel.kt

```diff
@@ -41,14 +48,19 @@ class GlobalSearchViewModel @Inject constructor(private val searchRepository: Se
         _queryText
                 .buffer(onBufferOverflow = BufferOverflow.DROP_OLDEST)
                 .mapLatest { query ->
+                    // Minimum search term is 2 characters
                     if (query.trim().length < 2) {
                         SearchResult.EMPTY
                     } else {
                         // user input delay here in case we get a new query within a few hundred ms
                         // this coroutine will be cancelled and expensive query will not be run if typing quickly
                         // first query of 2 characters will be instant however
                         delay(300)
-                        val settableFuture = SettableFuture<SearchResult>()
+
+                        // If we already have a search running then stop it
+                        if (!settableFuture.isDone) { settableFuture.cancel(true) }
```

Comment:
you don't need to check if the `Future` done, it's checked in `cancel`, the comment is not exactly true, as canceling this future would essentially stop listening for results, but won't stop any work being done.

Is it even necessary? `mapLatest` will prevent stale results clobbering newer ones.

Reply:
Cool, I didn't know `mapLatest` did that - have reverted `settableFuture` back to being a val defined at point of use